File: backend/app/schemas/document.py
import datetime
from typing import Optional, List, Dict, Any
from pydantic import BaseModel, Field

class DocumentBase(BaseModel):
    title: str
    category: str = "warranty"
    file_path: str
    file_name: str
    file_size: int = 0
    file_type: str = "application/pdf"
    issuer: Optional[str] = None
    document_date: Optional[str] = None
    expiry_date: Optional[str] = None
    warranty_months: Optional[int] = None
    contract_number: Optional[str] = None
    amount: Optional[float] = None
    physical_location: Optional[str] = None
    is_vault_protected: bool = False
    tags: Optional[str] = None
    summary: Optional[str] = None
    ocr_fulltext: Optional[str] = None
    related_entity_type: Optional[str] = None
    related_entity_id: Optional[int] = None


class DocumentCreate(DocumentBase):
    pass


class DocumentUpdate(BaseModel):
    title: Optional[str] = None
    category: Optional[str] = None
    file_path: Optional[str] = None
    file_name: Optional[str] = None
    file_size: Optional[int] = None
    file_type: Optional[str] = None
    issuer: Optional[str] = None
    document_date: Optional[str] = None
    expiry_date: Optional[str] = None
    warranty_months: Optional[int] = None
    contract_number: Optional[str] = None
    amount: Optional[float] = None
    physical_location: Optional[str] = None
    is_vault_protected: Optional[bool] = None
    tags: Optional[str] = None
    summary: Optional[str] = None
    ocr_fulltext: Optional[str] = None
    related_entity_type: Optional[str] = None
    related_entity_id: Optional[int] = None


class DocumentResponse(DocumentBase):
    id: int
    created_by_id: Optional[int] = None
    created_at: datetime.datetime
    updated_at: datetime.datetime
    days_until_expiry: Optional[int] = None
    status: str = "active"  # active, expiring_soon, expired, permanent

    class Config:
        from_attributes = True


class DocumentStatsResponse(BaseModel):
    total_documents: int
    expiring_soon_count: int
    expired_count: int
    vault_count: int
    categories: Dict[str, int]


class DocumentAiScanResponse(BaseModel):
    title: Optional[str] = None
    category: Optional[str] = "warranty"
    issuer: Optional[str] = None
    document_date: Optional[str] = None
    expiry_date: Optional[str] = None
    warranty_months: Optional[int] = None
    contract_number: Optional[str] = None
    amount: Optional[float] = None
    tags: Optional[str] = None
    summary: Optional[str] = None
    ocr_fulltext: Optional[str] = None


class VaultVerifyRequest(BaseModel):
    pin: str


class VaultSetPinRequest(BaseModel):
    old_pin: Optional[str] = None
    new_pin: str

File: frontend/src/types/document.ts
export type DocumentCategory =
  | 'warranty'
  | 'contract'
  | 'identity'
  | 'inspection'
  | 'manual'
  | 'medical'
  | 'housing'
  | 'vehicle'
  | 'other';

export type DocumentStatus = 'active' | 'expiring_soon' | 'expired' | 'permanent';

export interface DocumentItem {
  id: number;
  title: string;
  category: DocumentCategory;
  file_path: string;
  file_name: string;
  file_size: number;
  file_type: string;
  issuer?: string;
  document_date?: string;
  expiry_date?: string;
  warranty_months?: number;
  contract_number?: string;
  amount?: number;
  physical_location?: string;
  is_vault_protected: boolean;
  tags?: string;
  summary?: string;
  ocr_fulltext?: string;
  related_entity_type?: string;
  related_entity_id?: number;
  created_by_id?: number;
  created_at: string;
  updated_at: string;
  days_until_expiry?: number | null;
  status: DocumentStatus;
}

export interface DocumentCreate {
  title: string;
  category: DocumentCategory;
  file_path: string;
  file_name: string;
  file_size?: number;
  file_type?: string;
  issuer?: string;
  document_date?: string;
  expiry_date?: string;
  warranty_months?: number;
  contract_number?: string;
  amount?: number;
  physical_location?: string;
  is_vault_protected?: boolean;
  tags?: string;
  summary?: string;
  ocr_fulltext?: string;
  related_entity_type?: string;
  related_entity_id?: number;
}

export interface DocumentUpdate {
  title?: string;
  category?: DocumentCategory;
  file_path?: string;
  file_name?: string;
  file_size?: number;
  file_type?: string;
  issuer?: string;
  document_date?: string;
  expiry_date?: string;
  warranty_months?: number;
  contract_number?: string;
  amount?: number;
  physical_location?: string;
  is_vault_protected?: boolean;
  tags?: string;
  summary?: string;
  ocr_fulltext?: string;
  related_entity_type?: string;
  related_entity_id?: number;
}

export interface DocumentStats {
  total_documents: number;
  expiring_soon_count: number;
  expired_count: number;
  vault_count: number;
  categories: Record<string, number>;
}

export interface DocumentAiScanResponse {
  title?: string;
  category?: DocumentCategory;
  issuer?: string;
  document_date?: string;
  expiry_date?: string;
  warranty_months?: number;
  contract_number?: string;
  amount?: number;
  tags?: string;
  summary?: string;
  ocr_fulltext?: string;
}

export interface DocumentUploadResult {
  file_path: string;
  file_name: string;
  file_size: number;
  file_type: string;
  ai_metadata?: DocumentAiScanResponse;
}
